rmgpy/rmg/react.py

```diff
@@ -239,6 +239,36 @@ def convertToSpeciesObjects(reaction):
     except AttributeError:
         pass
 
+
+def correctDegeneracyOfReverseReactions(reactionList, reactants):
+    """
+    This method corrects the degeneracy of reactions found when the backwards
+    template is used. Given the following parameters:
+
+        reactionList - list of reactions with their degeneracies already counted
+        reactants - list/tuple of species used in the generateReactions method
+
+    This method modifies reactionList in place and returns nothing
+    
+    This does not adjust for identical reactants, you need to use `reduceSameReactantDegeneracy`
+    to adjust for that.
+    """
+    from rmgpy.reaction import _isomorphicSpeciesList
+    from rmgpy.reaction import ReactionError
+
+    for rxn in reactionList:
+        if _isomorphicSpeciesList(rxn.reactants, reactants):
```

Comment:
Is there a reason we can't add some boolean property to the reaction object during rate estimation (such as "fromReverse") so we already know if it's the reverse template or not and we don't have to do all of these isomorphism checks?